python/openvino_tokenizers/tokenizer_transformations.py

```diff
@@ -1,250 +1,197 @@
-        str_unpack_pattern = WrapType("opset15.StringTensorUnpack", [str_input_pattern])
-
-        def callback(matcher: Matcher) -> bool:
-            str_unpack: ov.Node = matcher.get_match_root()
-
-            new_parameter = op.Parameter(Type.string, PartialShape([-1, -1]))
-            str_unpack.input(0).replace_source_output(new_parameter.output(0))
-            nonlocal model
-            model.replace_parameter(0, new_parameter)
+    def run_on_model(self, model: ov.Model):
+        parameters = model.get_parameters()
+        if len(parameters) != 1:
+            return False
 
-            inp_shape = opset.shape_of(new_parameter, output_type="i32")
-            num_batches = opset.slice(inp_shape, [0], [1], [1])
-            boradcasted_shape = opset.concat(
-                [num_batches, make_constant_node([number_of_inputs], Type.i32)], axis=0, name="broadcasted_shape"
-            )
-
-            # Save for future use to prevent loops
-            target_begins = str_unpack.output(0).get_target_inputs()
-            target_ends = str_unpack.output(1).get_target_inputs()
+        target_inputs = list(parameters[0].output(0).get_target_inputs())
+        str_unpack: ov.Node = target_inputs[0].get_node()
+        if str_unpack.get_type_name() != "StringTensorUnpack":
+            return False
 
-            begins = opset.broadcast(str_unpack.output(0), boradcasted_shape)
-            ends = opset.broadcast(str_unpack.output(1), boradcasted_shape)
+        # Replace single Parameter with concatenated pair of Parameters.
+        new_parameters = [ov.op.Parameter(Type.string, PartialShape([-1])) for i in range(2)]
+        new_input = opset.concat(new_parameters, axis=0)
+        str_unpack.input(0).replace_source_output(new_input.output(0))
 
-            num_inputs_node = opset.slice(inp_shape, [1], [2], [1])
-            equal = opset.equal(num_inputs_node, make_constant_node(1, Type.i32), name="is_paired_input")
-
-            # If the number of inputs is 1, we need to zero the second dimension
-            # of the broadcasted begins and ends tensors.
-            # This is done so that the second input string is empty "".
-            multiplier = opset.select(
-                equal, make_constant_node([[1, 0]], Type.i32), make_constant_node([[1, 1]], Type.i32)
-            )
-            begins = opset.multiply(begins, multiplier)
-            ends = opset.multiply(ends, multiplier)
+        model.replace_parameter(0, new_parameters[0])
+        model.add_parameters([new_parameters[1]])
 
-            begins_ = opset.reshape(begins, make_constant_node([-1], Type.i32), special_zero=False)
-            ends_ = opset.reshape(ends, make_constant_node([-1], Type.i32), special_zero=False)
+        combine_seg = None
+        for op in model.get_ops():
+            if op.get_type_name() == "CombineSegments":
+                combine_seg = op
+        if not combine_seg:
+            return False
```

Comment:
may be this check is better to perform before any creation of new Parameter and adding it to the model?

Reply:
Refactored this code. Now we don't add new nodes unless we ensured that the old ov::Model has required structure.